arb_os/codeSegment.mini

```diff
@@ -116,15 +117,22 @@ impure func parseEvmCodeSegment(bs: ByteStream) -> EvmInstructionStack {
             if ((opcode >= 0x60) && (opcode <= 0x7f)) {
                 // it's a pushN instruction; unmarshal the immediate value and push it
                 let bytesRemaining = opcode - 0x5f;
-                let imm = 0;
-                while (bytesRemaining > 0) {
-                    if let Some(res) = bytestream_getByte(bs) {
-                        let (ubs, val) = res;
-                        bs = ubs;
-                        imm = 256*imm + val;
+                let imm = xif let Some(res) = bytestream_getPartialWord(bs, bytesRemaining) {
+                    let (ubs, v) = res;
+                    bs = ubs;
+                    v
+                } else {
+                    let val = 0;
+                    while (bytesRemaining > 0) {
+                        if let Some(res) = bytestream_getByte(bs) {
+                            let (ubs, v) = res;
+                            bs = ubs;
+                            val = 256*val + v;
+                        }
+                        bytesRemaining = bytesRemaining-1;
                     }
```

Comment:
Is is possible to hit this case?

Reply:
There is a corner case where that could happen, when the input byte stream ends unexpectedly before getting to the end of the expected immediate value length.  That does happen in practice, when we're parsing non-code data at the end of a file (without realizing that it's not code).

Reply:
👍